client/src/components/CommentApp/comments.js

```diff
@@ -0,0 +1,144 @@
+import { initCommentsApp } from 'wagtail-comment-frontend';
+import { STRINGS } from '../../config/wagtailConfig';
+
+function initComments(element, author, initialComments) {
+  const app = initCommentsApp(element, author, initialComments, STRINGS);
+  window.commentApp = app;
+  return app;
+}
+
+function getContentPath(fieldNode) {
+  // Return the total contentpath for an element as a string, in the form field.streamfield_uid.block...
+  if (fieldNode.closest('data-contentpath-disabled')) {
+    return '';
+  }
+  let element = fieldNode.closest('[data-contentpath]');
+  const contentPaths = [];
+  while (element !== null) {
+    contentPaths.push(element.dataset.contentpath);
+    element = element.parentElement.closest('[data-contentpath]');
+  }
+  contentPaths.reverse();
+  return contentPaths.join('.');
+}
```

Comment:
So I do agree with you, I'm not too fond of this either. Your suggestion was my original thought too, and works fine for plain fields. However, the problem is with newly initialised widgets, particularly within `StreamField`. Right now, they aren't very intelligent - they don't know about their context, and since they're added clientside, they don't have an opportunity to get their paths in python. This was a workaround to ensure a widget could pick up a contentpath wherever it was. One thought I had is that we might be able to replace this when Matthew and Karl are further into their Telepath work, such that a streamfield block could more intelligently hand a base contentpath over to its child and say "okay, add your contentpath onto the end and initialise your widget". If you have any thoughts on alternatives for streamfield though, I'm interested!

(The translation app doesn't modify the page editor in the same way, so no)

Reply:
I've discussed this with Karl, and suggested potentially including the concept of contentPath for streamfield blocks as part of the Telepath sprint, so hopefully this can be a short term measure, but neither of us could come up with another workaround within the existing system.